### PR TITLE
No computer matching agreements

### PR DESCRIPTION
**Pull request by ondrae**
This PR stops including the computer matching agreements. Our user research has told us to not include them in our service.

While writing this PR, I discovered we had a bunch of computer matching agreements we didn't know about. We were only checking the `action` for them, yet there were hundreds that had nondescript `action` but had a `title` saying it was a computer matching agreement.

We now check both `title` and `action` and exclude any matching agreements.

I've done a full `find_sorns` job locally with this code and it worked.
### Notes
After deploying, we should delete existing CMAs from our current servers and local machines.
```
Sorn.where('action ILIKE ?', '%match%').destroy_all
Sorn.where('title ILIKE ?', '%match%').destroy_all
```

### CODE REVIEW

File: app/models/federal_register_client.rb [hunk 1]
```diff
@@ -94,6 +81,27 @@ def sorn_params(result)
...
+    # exclude any computer matching agreements
+    result.type == 'Notice' && a_sorn_title?(result.title) && not_a_cma?(result)
+  end
+
+  def a_sorn_title?(title)
+    # titles with either privacy act or system of record
+    title.match? /(privacy act|system[\ssof]*record)/i
+  end
+
+  def not_a_cma?(result)
```
Comment (ondrae) [inline app/models/federal_register_client.rb:98]: The main change of this PR

File: app/models/federal_register_client.rb [hunk 2]
```diff
@@ -94,6 +81,27 @@ def sorn_params(result)
...
+    # find the expected sorn titles
+    # exclude any computer matching agreements
+    result.type == 'Notice' && a_sorn_title?(result.title) && not_a_cma?(result)
+  end
+
+  def a_sorn_title?(title)
+    # titles with either privacy act or system of record
+    title.match? /(privacy act|system[\ssof]*record)/i
+  end
+
```
Comment (ondrae) [inline app/models/federal_register_client.rb:97]: The main change of this PR. This method looks for `match` in either the title or the action.

File: app/models/federal_register_client.rb (hunk 1 above)
Comment (ondrae) [inline app/models/federal_register_client.rb:98]: The main change of this PR. This method looks for `match` in either the title or the action.

File: app/models/federal_register_client.rb [hunk 3]
```diff
@@ -94,6 +81,27 @@ def sorn_params(result)
...
+  def not_a_cma?(result)
+    # no titles or actions with the word match
+    not_in_title = result.title.downcase.exclude?('match')
+    not_in_action = result.action.downcase.exclude?('match') if result.action.present?
+    not_in_title && not_in_action
```
Comment (ondrae) [inline app/models/federal_register_client.rb:102]: The main change of this PR. This method looks for `match` in either the title or the action.

File: app/models/federal_register_client.rb [hunk 4]
```diff
@@ -104,8 +112,6 @@ def add_agencies(result, sorn)
     end
   end
 
-  private
```
Comment (ondrae) [inline app/models/federal_register_client.rb:107]: We had private declared twice accidentally.

**Review by lauraGgit — APPROVED** [review]
looks good!

**Review by peterrowland — APPROVED** [review]
√

**Pull request merged**